### PR TITLE
Get metadata from central repo

### PR DESCRIPTION
**Pull request by lechnerc77**
Read metadata from central repo

### CODE REVIEW

File: libs/javascript/metadatafetcher.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    const result = await getContentFromGitHub(sourcePath);
+
+    await storeResultInRepo(result, targetPath);
+}
+
+async function getContentFromGitHub(sourcePath) {
+
+    const pat = process.env["METADATA_READ_TOKEN"];
+    const owner = "btp-setup-automator";
+    const repo = "btp-metadata-fetcher";
```
Comment (ublock) [inline libs/javascript/metadatafetcher.js:33]: Should this information be public? If not, it should be provided via environmental.

File: libs/javascript/package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "main": "metadatafetcher.js",
+  "scripts": {
+    "fetch:metadata": "node metadatafetcher.js"
+  },
+  "author": "Christian Lechner",
+  "license": "Apache2",
+  "dependencies": {
+    "@octokit/auth-token": "3.0.0",
+    "@octokit/request": "6.2.0",
+    "node-fetch": "2.6.7"
```
Comment (ublock) [inline libs/javascript/package.json:14]: Pinning and reproducible builds are covered by package-lock + "npm ci"; here we can use an open range:

```suggestion
    "@octokit/auth-token": "^3.0.0",
    "@octokit/request": "^6.2.0",
    "node-fetch": "^2.6.7"
```

File: libs/javascript/metadatafetcher.js (hunk 1 above)
Comment (lechnerc77) [reply libs/javascript/metadatafetcher.js:33]: Makes sense to parameterize these parameters

File: libs/javascript/metadatafetcher.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+const { createTokenAuth } = require("@octokit/auth-token");
+const fetch = require('node-fetch');
+const fs = require('node:fs');
+const { mkdir } = require('node:fs/promises');
+const path = require('node:path');
+const { request } = require("@octokit/request");
+
+async function createDirIfNotExisting(dirPath) {
+
+    if (!fs.existsSync(dirPath)) {
```
[the remark this answers is not in the document]
Comment (lechnerc77) [reply libs/javascript/metadatafetcher.js:12]: As discussed - for this scenario we keep it as is. Certainly worth a rewrite in a later stage

File: libs/javascript/package.json (hunk 2 above)
Comment (lechnerc77) [reply libs/javascript/package.json:14]: Keeping the hard pinned deps in package.json

**Pull request merged**